srcpkgs/python3-txredisapi/template

```diff
@@ -0,0 +1,15 @@
+# Template file for 'python3-txredisapi'
+pkgname=python3-txredisapi
+version=1.4.7
+revision=1
+wrksrc="txredisapi-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-Twisted python3-six"
+checkdepends="$depends"
```

Comment:
Could you test if tests work when with python3-pytest?

Reply:
Yep, indeed it does work with pytest, thanks.

Reply:
Then we should add python3-pytest to checkdepends. xbps-src runs tests with pytest when pytest is installed, otherwise it falls back to setup.py.

Reply:
Ah whoops, I thought you were asking about synapse here, which did work with pytest (unlike the last time I tried). python3-txredisapi does not work with pytest, I just checked.

Reply:
Actually, I don't think any tests are included in the PyPi source. With distfiles directly from GitHub, tests do seem to work, I'll push an update for that in a little bit.

Reply:
It might be too complex to add, txredisapi needs a running instance of Redis in order to test. I tried the naive approach:
```
do_check() {
  redis-server &
  python3 -m pytest
  redis-cli flushall
}
```
but I'm getting a bunch of errors about type mismatches:
```
self = <tests.test_unix_connection.TestUnixConnectionMethods testMethod=test_UnixConnectionDB1>
result = <TestCaseFunction test_UnixConnectionDB1>

    def run(self, result):
        """
        Run the test case, storing the results in C{result}.

        First runs C{setUp} on self, then runs the test method (defined in the
        constructor), then runs C{tearDown}.  As with the standard library
        L{unittest.TestCase}, the return value of these methods is disregarded.
        In particular, returning a L{Deferred<twisted.internet.defer.Deferred>}
        has no special additional consequences.

        @param result: A L{TestResult} object.
        """
        log.msg("--> %s <--" % (self.id()))
        new_result = itrial.IReporter(result, None)
        if new_result is None:
            result = PyUnitResultAdapter(result)
        else:
            result = new_result
        result.startTest(self)
        if self.getSkip(): # don't run test methods that are marked as .skip
>           result.addSkip(self, self.getSkip())
E           TypeError: Skipped expected string as 'msg' parameter, got 'bool' instead.
E           Perhaps you meant to use a mark?

/usr/lib/python3.10/site-packages/twisted/trial/_synctest.py:1062: TypeError
```

Reply:
ok, thanks for investigating, in that case add with make_check=no with an appropriate comment

Reply:
Nevermind, I got it working by skipping those tests, since Redis doesn't have a Unix socket enabled by default.

Reply:
I think you didn't want to commit dump.rdb

Reply:
🤦‍♂️thanks, removed. I really need to stop relying on `git add -A`...